app/src/main/java/fr/coppernic/scansample/MainActivity.java

```diff
@@ -50,6 +64,44 @@ public void onClick(View view) {
                 }
             }
         });
+        btnStartStop = findViewById(R.id.btnStartStop);
+        btnStartStop.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                //ask permission for Android 7 and upper
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
```

Comment:
android 8 and upper!

Reply:
N pour nougat je fais les demandes d'autorisation a partir de l'api 24